src/ln/channelmanager.rs

```diff
@@ -685,25 +694,40 @@ impl ChannelManager {
 						Ok(funding_msg) => {
 							(chan, funding_msg.0, funding_msg.1)
 						},
-						Err(_e) => {
-							//TODO: Push e to pendingevents
+						Err(e) => {
+							println!("Got error handling message: {}!", e.err);
+							if let Some(action) = e.msg {
+								match action {
+									msgs::ErrorAction::DisconnectPeer {} => {
```

Comment:
Hmmm, I'm really not a fan of this setup - needing to handle conversion of errors to different types of errors at each place where we can generate outbound messages really sucks. Just having a general HandleOutboundMsgGenerationError event which looks like SendErrorMessage as proposed but instead of msg, disconnect, and hard, it just holds a HandleError should simplify this PR, and more importantly, future PRs. Further, SendErrorMessage would be a HandleError::msg action option (with a disconnect option, or maybe it could go in DisconnectPeer? I'm not picky there, I think).